### PR TITLE
feat(log-store): add metrics for log store

### PR DESCRIPTION
**Pull request by wenym1**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

Add metrics for sink and log store.

Dashboard with newly added metrics is as followed:
image

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

File: src/stream/src/executor/monitor/streaming_stats.rs [hunk 1]
```diff
@@ -751,6 +762,78 @@ impl StreamingMetrics {
         )
         .unwrap();
 
+        let log_store_first_write_epoch = register_int_gauge_vec_with_registry!(
```
Comment (hzxa21) [inline src/stream/src/executor/monitor/streaming_stats.rs:765]: All the newly added metrics are bound to executor_id, do we drop them when the actor/executor is dropped?

Comment (wenym1) [reply src/stream/src/executor/monitor/streaming_stats.rs]: I implemented a `LabelGuardedXxx`, which will remove the label when it is dropped.

**Review by hzxa21 — APPROVED** [review]

File: src/stream/src/executor/sink.rs [hunk 2]
```diff
@@ -302,18 +300,17 @@ impl<F: LogStoreFactory> SinkExecutor<F> {
             .filter_map(|(idx, column)| (!column.is_hidden).then_some(idx))
             .collect_vec();
 
-        let log_reader = log_reader.transform_chunk(move |chunk| {
-            sink_metrics
-                .connector_sink_rows_received
-                .inc_by(chunk.cardinality() as u64);
```
Comment (ZENOTME) [inline src/stream/src/executor/sink.rs:308]: Is this metrics been missed now? cc @wenym1

Comment (ZENOTME) [reply src/stream/src/executor/sink.rs:308]: I find that the metrics in Dashboard is always 0.
image

Comment (chenzl25) [reply src/stream/src/executor/sink.rs:308]: Yes. This metric has been removed in this PR. Let's add it back https://github.com/risingwavelabs/risingwave/pull/12812

**Pull request merged**